src/net_processing.cpp

```diff
@@ -488,6 +488,14 @@ struct Peer {
     /** Whether this peer should be disconnected and marked as discouraged (unless it has the noban permission). */
     bool m_should_discourage GUARDED_BY(m_misbehavior_mutex){false};
 
+    /** Protects block inventory data members */
+    RecursiveMutex m_block_inv_mutex;
+    /** List of block ids we still have announce.
+     * There is no final sorting before sending, as they are always sent
+     * immediately and in the order requested. */
+    std::vector<uint256> vInventoryBlockToSend GUARDED_BY(m_block_inv_mutex);
+    /** List of headers to relay */
```

Comment:
`vBlockHashesToAnnounce` is a list of block hashes not headers ? With regards to renaming follow-up commit, I would lean towards the most verbose option `m_blockids_for_inv_relay/m_blockids_for_headers_relay` to denotate that these vectors contain an identifier even if the endgoal is to relay the identifiee (either block or headers).

Reply:
Ok, I'll update the comment to "List of block ids to announce headers for".

> I would lean towards the most verbose option m_blockids_for_inv_relay/m_blockids_for_headers_relay

Sure. I'll update.